src/phpDocumentor/Reflection/Php/Factory/Class_.php

```diff
@@ -0,0 +1,153 @@
+     * used to create nested Elements.
+     *
+     * @param ClassNode $object object to convert to an Element
+     * @param StrategyContainer $strategies used to convert nested objects.
+     * @return ClassDescriptor
+     */
+    public function create($object, StrategyContainer $strategies)
+    {
+        if (!$this->matches($object)) {
+            throw new InvalidArgumentException(
+                sprintf('%s cannot handle objects with the type %s',
+                    __CLASS__,
+                    is_object($object) ? get_class($object) : gettype($object)
+                )
+            );
+        }
+
+        $docBlock = $this->createDocBlock($object->getDocComment(), $strategies);
+
+        $classDescriptor = new ClassDescriptor(
+            new Fqsen('\\' . $object->name),
+            $docBlock,
+            $object->extends ? new Fqsen('\\' . $object->extends) : null,
+            $object->isAbstract(),
+            $object->isFinal()
+        );
+
+        if (isset($object->implements)) {
+            foreach ($object->implements as $interfaceClassName) {
+                $classDescriptor->addInterface(
+                    new Fqsen('\\' . $interfaceClassName->toString())
+                );
+            }
+        }
+
+        if (isset($object->stmts)) {
+            foreach ($object->stmts as $stmt) {
+                switch (get_class($stmt)) {
+                    case TraitUse::class:
+                        foreach ($stmt->traits as $use) {
+                            $classDescriptor->addUsedTrait(new Fqsen('\\'. $use->toString()));
+                        }
+                        break;
+                    case PropertyNode::class:
+                        $properties = new PropertyIterator($stmt);
+                        foreach ($properties as $property) {
+                            $this->addCreateAndMember($property, $strategies, $classDescriptor);
+                        }
+                        break;
+                    case ClassConst::class:
+                        $constants = new ClassConstantIterator($stmt);
+                        foreach ($constants as $const) {
+                            $this->addCreateAndMember($const, $strategies, $classDescriptor);
+                        }
+                        break;
+                    default :
+                        $this->addCreateAndMember($stmt, $strategies, $classDescriptor);
+                        break;
+                }
+            }
+        }
+
+        return $classDescriptor;
+    }
+
+    /**
+     * @param Node|PropertyIterator|ClassConstantIterator $stmt
+     * @param StrategyContainer $strategies
+     * @param ClassDescriptor $classDescriptor
+     */
+    private function addCreateAndMember($stmt, StrategyContainer $strategies, ClassDescriptor $classDescriptor)
+    {
+        $strategy = $strategies->findMatching($stmt);
```

Comment:
Wouldn't it be better to move line 123 and 124 to its own method and move the addMethod/Property/Constant calls back to 97, 103 and a specific version for Method there? That saves us a switch statement and a form of duplication